plugins/item_previews/web_client/js/item_previews.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * The Item Preview widget shows a preview of items under a given folder.
+ * For now, the only supported item previews are image previews.
+ */
+
+var MAX_JSON_SIZE = 2e6 /* bytes */;
+
+var    _isSupportedItem = function (item) {
+  return _isImageItem(item) || _isJSONItem(item);
+};
+
+var _isJSONItem = function (item) {
+  return /\.(json)$/i.test(item.get('name'));
+};
+
+var _isImageItem = function (item) {
+  return /\.(jpg|jpeg|png|gif)$/i.test(item.get('name'));
+};
```

Comment:
These four symbols will be placed in global scope, which we generally try to avoid. Since they seem to only be referenced inside `ItemPreviewWidget`, let's move them as member variables within that prototype :)